avm/res/network/virtual-network/main.bicep

```diff
@@ -111,23 +111,23 @@ resource virtualNetwork 'Microsoft.Network/virtualNetworks@2023-04-01' = {
       name: subnet.name
       properties: {
         addressPrefix: subnet.addressPrefix
-        addressPrefixes: contains(subnet, 'addressPrefixes') ? subnet.addressPrefixes : []
-        applicationGatewayIPConfigurations: contains(subnet, 'applicationGatewayIPConfigurations') ? subnet.applicationGatewayIPConfigurations : []
-        delegations: contains(subnet, 'delegations') ? subnet.delegations : []
-        ipAllocations: contains(subnet, 'ipAllocations') ? subnet.ipAllocations : []
-        natGateway: contains(subnet, 'natGatewayResourceId') ? {
+        addressPrefixes: contains(subnet, 'addressPrefixes') && !empty(subnet.addressPrefixes) ? subnet.addressPrefixes : []
```

Comment:
Hey there @SvenAelterman, thank you for your contribution.
I'm wondering though, can you help me understand how this changes the behavior.

After
- If a user does provide addressPrefixes (not empty) it will set them.
- f the user does not provide addressPrefixes it uses `[]`

Before
- If a user does provide addressPrefixes it will set them (empty or not).
- f the user does not provide addressPrefixesit uses `[]`

I'd anyways suggest to update these properties to use the `tryGet` via
`addressPrefixes: subnet.?addressPrefixes : []`

Reply:
There isn't really a change in behavior. But `contains(object, 'property_name')` checks if the property exists but not if the value of the property is something valid that ARM will accept.

I have some code of my own where I am trying to leverage AVM now where subnet definitions are somewhat dynamically generated. This means that in some cases, properties like a network security group ID are present but could be `null`.

Reply:
Gotcha, in that case, `addressPrefixes: subnet.?addressPrefixes ?? []` should be a great (& shorter) compromise.

In case you have not seen this before let me quickly describe what the syntax does:
- The `subnet.?addressPrefixes` safely accesses the property 'addressPrefixes' in the object 'subnet' in that does does not fail if it is not set at all. If it would not be set, the value would simply default to `null`
- The `?? []` in the end is a snippet that returns the first non-null value. For example `null ?? null ?? 'hey there'` would return 'hey there'. In our case, what it does is defaulting to `[]`

So the combined example essentially uses the value of `addressPrefixes` if it was provided or `[]` if not. 

Now, IF `[]` is not desired as a default and property accepts `null`, `addressPrefixes: subnet.?addressPrefixes` would be the even better solution.

Sidenote: When using `addressPrefixes: subnet.?addressPrefixes` when referencing another module, Bicep will actually go and use the default value that was defined the the referenced module for its parameter `addressPrefixes` - which is really neat. Not immediately relevant now, but I figured worth mentioning :)

Reply:
I am familiar with that syntax from C#, but I didn't know it was available in Bicep. I will update the PR.

Reply:
I can't recall when it was added - but fortunately sometime mid-last year, I think :)

Reply:
Hi @SvenAelterman , could you update the code as @AlexanderSehr was mentioning? Thank you very much

Reply:
I will, I have just not had time to revisit this.

Reply:
Just checking in 😉